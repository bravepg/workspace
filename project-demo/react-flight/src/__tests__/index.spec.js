import React, { Component } from 'react'
import { mount, shallow } from 'enzyme'
import Director from '../core/director'
import VelocityComposer from '../core/velocity-composer'
import Flight, { Rect, Oval } from '../dom/index'
window.requestAnimationFrame = f => f()

class Comp extends Component {
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <Flight ref="flight">
        <Flight.Frame source>
          <div>
            <div
              id="line1"
              className="line1"
              radius={5}
              style={{
                backgroundColor: '#95A2AA',
                left: 120,
                top: 100 - 40,
                width: 130,
                height: 10,
              }}
            />

            <Rect
              name={'line2'}
              radius={5}
              style={{
                backgroundColor: '#E6ECF0',
                left: 95,
                top: 115 - 40,
                width: 180,
                height: 10,
              }}
            />

            <Oval
              name="circ1"
              size={16}
              style={{
                backgroundColor: '#79CD15',
                left: 110,
                top: 130 - 40,
              }}
            />
          </div>
        </Flight.Frame>
        <Flight.Frame>
          <div>
            <div
              className="line1"
              radius={5}
              style={{
                backgroundColor: '#95A2AA',
                left: 777,
                top: 100 - 40,
                width: 130,
                height: 10,
              }}
            />

            <Rect
              name={'line2'}
              radius={5}
              style={{
                backgroundColor: '#E6ECF0',
                left: 0,
                top: 115 - 40,
                width: 180,
                height: 10,
              }}
            />

            <Oval
              name="circ1"
              size={16}
              style={{
                backgroundColor: '#79CD15',
                left: 110,
                top: 130 - 40,
              }}
            />
          </div>
        </Flight.Frame>
      </Flight>
    )
  }
}

describe('flight', () => {
  it('integration', () => {
    const c = mount(<Comp />)
    const d = c.ref('flight').getNode().state.director
    const movements = JSON.stringify(d.movements, (k, v) => {
      if (k === 'root' || k === 'ref') {
        return '-test-'
      }
      return v
    })

    // Flight.Frame `source` prop, should signal Flight components to render DOM ids
    expect(c.find('#circ1').exists()).toEqual(true)
    expect(c.find('#line2').exists()).toEqual(true)

    // capture movements
    expect({
      frames: d.frames,
      movements,
      timelines: d.composer.timelines.length,
    }).toMatchSnapshot()
  })
})
