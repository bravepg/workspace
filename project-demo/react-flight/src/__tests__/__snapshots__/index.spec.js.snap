exports[`flight integration 1`] = `
Object {
  "frames": Array [
    Object {
      "duration": 500,
    },
    Object {
      "duration": 500,
    },
  ],
  "movements": "[[{\"root\":\"-test-\",\"target\":{\"style\":{\"backgroundColor\":\"#95A2AA\",\"left\":777,\"top\":60,\"width\":130,\"height\":10},\"ref\":\"-test-\",\"name\":\"line1\"},\"duration\":500},{}],[{\"root\":\"-test-\",\"target\":{\"style\":{\"borderRadius\":5,\"backgroundColor\":\"#E6ECF0\",\"left\":0,\"top\":75,\"width\":180,\"height\":10},\"ref\":\"-test-\",\"name\":\"line2\"},\"duration\":500},{}],[{\"root\":\"-test-\",\"target\":{\"style\":{\"borderRadius\":8,\"width\":16,\"height\":16,\"backgroundColor\":\"#79CD15\",\"left\":110,\"top\":90},\"ref\":\"-test-\",\"name\":\"circ1\"},\"duration\":500},{}]]",
  "timelines": 3,
}
`;
